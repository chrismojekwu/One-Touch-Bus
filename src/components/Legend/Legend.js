import React from 'react'
import User from '../../img/marker-icon.png'
import Bus from '../../img/bus-icon.png'

function Legend() {
        return (
            <section className="legend">
                 <img className="legend-icon" src={User} alt="Icon for the users positon."/>Your Current Position
                 <img className="legend-icon" src={Bus} alt="Icon for any nearby buses."/>Bus (Click for Route Name and Destination)
            </section>
        )
}

export default Legend