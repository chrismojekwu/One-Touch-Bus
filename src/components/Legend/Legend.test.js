import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from 'react-test-renderer';
import Legend from './Legend';

it('renders the legend component and marker imgs without crashing', () => {
    render(<Legend/>);
    expect(screen.getAllByRole('img').length).toBe(2);
})

it('renders snapshot correcrtly', () => {
    const tree = 
    renderer.create(<Legend/>).toJSON();
    expect(tree).toMatchSnapshot();
});