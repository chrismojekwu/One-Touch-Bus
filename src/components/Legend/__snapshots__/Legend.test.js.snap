// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correcrtly 1`] = `
<section
  className="legend"
>
  <img
    alt="Icon for the users positon."
    className="legend-icon"
    src="marker-icon.png"
  />
  Your Current Position
  <img
    alt="Icon for any nearby buses."
    className="legend-icon"
    src="bus-icon.png"
  />
  Bus (Click for Route Name and Destination)
</section>
`;
